sdk/src/transaction_context.rs

```diff
@@ -139,29 +140,32 @@ impl TransactionContext {
         self.account_keys.iter().rposition(|key| key == pubkey)
     }
 
+    /// Returns instruction trace length
+    pub fn get_instruction_trace_length(&self) -> usize {
+        self.instruction_trace.len()
+    }
+
+    /// Gets an InstructionContext by its index in the trace
+    pub fn get_instruction_context_at_index(
```

Comment:
nit: I find this method name to be difficult to understand. I wouldn't guess from its name that it's indexing into a flattened instruction trace. I think I prefer the way the previous code would get the trace and then index into the returned value.

Reply:
See https://github.com/solana-labs/solana/pull/27109#discussion_r949619600 for why this interface changes in this fashion. Yes, the method might need a better name (proposals welcome).

Reply:
Hmm maybe just change `at_index` to `at_trace_index`? Another thing that might be interesting is having a `get_instruction_trace()` method that returns at iterator over instruction contexts